parsons/google/google_drive.py

```diff
@@ -0,0 +1,153 @@
+import os
+import json
+import logging
+
+from parsons.google.utitities import setup_google_application_credentials
+from googleapiclient.discovery import build
+from google.oauth2.service_account import Credentials
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleDrive:
+    """
+    A connector for Google Drive, largely handling permissions.
+
+    `Args:`
+        google_keyfile_dict: dict
+            A dictionary of Google Drive API credentials, parsed from JSON provided
+            by the Google Developer Console. Required if env variable
+            ``GOOGLE_DRIVE_CREDENTIALS`` is not populated.
+        subject: string
+            In order to use account impersonation, pass in the email address of the account to be
+            impersonated as a string.
+    """
+
+    def __init__(self, google_keyfile_dict=None, subject=None):
+
+        scope = ["https://www.googleapis.com/auth/drive"]
+
+        setup_google_application_credentials(
+            google_keyfile_dict, "GOOGLE_DRIVE_CREDENTIALS"
+        )
+        google_credential_file = open(os.environ["GOOGLE_DRIVE_CREDENTIALS"])
+        credentials_dict = json.load(google_credential_file)
+
+        credentials = Credentials.from_service_account_info(
+            credentials_dict, scopes=scope, subject=subject
+        )
```

Comment:
Would you consider using the new pattern adopted by other google connectors as of #1040 to avoid reimplementing the issue #1039?

Reply:
@austinweisgrau I previously had:
```
self.client = build("drive", "v3", credentials=credentials)
```

If I follow your code from the other PR, will the following work too:
```
self.client = AuthorizedSession(credentials)
```